### PR TITLE
Fix #1571 Delete all log option added

### LINKED ISSUE #1571: Option to delete all the logged data at once
**Actual Behaviour**

No option to delete all the log entries
**Expected Behaviour**

There should be an option to delete all the log data

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1571 

**Changes**: Added new menu option in data logger activity to delete all the logged data

**Screenshot/s for the changes**: 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [ ] I have requested reviews from other members

### CODE REVIEW

**Comment by harsh-2711** [comment]
@neel1998 Please squash all your commits

**Comment by neel1998** [comment]
@harsh-2711  kindly review

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1093,5 +1093,6 @@
     <string name="no_data_fetched">No Data Fetched</string>
     <string name="recorder">Recorder</string>
     <string name="save_graph">Save Graph</string>
-
+    <string name="delete_all">Delete All Data</string>
+    <string name="delete_all_message">Are Sure You want Delete All Logged Data?</string>
```
Comment (harsh-2711) [inline app/src/main/res/values/strings.xml]: @neel1998 Follow standard casing of letters. Else the PR looks good.

Comment (neel1998) [reply app/src/main/res/values/strings.xml]: you mean make the 'w' of 'want' upper case, right?

Comment (harsh-2711) [reply app/src/main/res/values/strings.xml]: Change it too - "Are you sure wanna delete all logged data?" Also change the line above it.

Comment (neel1998) [reply app/src/main/res/values/strings.xml]: Aren't we keeping first character of each word in upper case through out the project?

Comment (neel1998) [reply app/src/main/res/values/strings.xml]: @harsh-2711  ?

Comment (CloudyPadmal) [reply app/src/main/res/values/strings.xml]: Oops. We need to fix that issue. Sentences need to be grammatically correct. Could you work on thay @neel1998?

Comment (neel1998) [reply app/src/main/res/values/strings.xml]: @CloudyPadmal  so the string should be "Are You Sure You Want To Delete All Logged Data?"

Comment (neel1998) [reply app/src/main/res/values/strings.xml]: @harsh-2711  can you please explain what exactly needs to be changed so that this PR can be closed

Comment (cweitat) [reply app/src/main/res/values/strings.xml]: "Are you sure?"

Is enough.

**Comment by mariobehling** [comment]
Please finalize this PR.

**Comment by neel1998** [comment]
@mariobehling  kindly review and merge

**Comment by neel1998** [comment]
@mariobehling  kindly review this PR, since it has been pending for a long. Thanks

File: app/src/main/java/io/pslab/activity/DataLoggerActivity.java [hunk 2]
```diff
@@ -86,11 +100,60 @@ public void onBackPressed() {
...
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.log_activity_menu, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+    private class DeleteAllTask extends AsyncTask<Void, Void, Void>{
+        @Override
+        protected Void doInBackground(Void... voids) {
+            Realm realm = Realm.getDefaultInstance();
+            for (SensorDataBlock data : realm.where(SensorDataBlock.class)
+                    .findAll().sort("block", Sort.DESCENDING)){
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/activity/DataLoggerActivity.java]: Do we need to sort when deleting?

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1093,5 +1093,6 @@
     <string name="no_data_fetched">No Data Fetched</string>
     <string name="recorder">Recorder</string>
     <string name="save_graph">Save Graph</string>
-
+    <string name="delete_all">Delete All Data</string>
+    <string name="delete_all_message">Are Sure You?</string>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Are you sure?

Comment (neel1998) [reply app/src/main/res/values/strings.xml]: xD my Bad. I will fix it

File: app/src/main/java/io/pslab/activity/DataLoggerActivity.java (hunk 2 above)
Comment (neel1998) [reply app/src/main/java/io/pslab/activity/DataLoggerActivity.java]: I will check once

**Comment by neel1998** [comment]
@CloudyPadmal @mariobehling  @harsh-2711  @cweitat  kindly review

**Comment by neel1998** [comment]
@CloudyPadmal @harsh-2711 this can be merged

**Comment by Avjeet** [comment]
@neel1998 Can u plz provide the link to the final apk for this PR for testing.

**Comment by neel1998** [comment]
[delete_all.apk.zip](https://github.com/fossasia/pslab-android/files/3094748/delete_all.apk.zip)
@Avjeet  here is the zip file for apk

**Comment by neel1998** [comment]
@Avjeet  i will look into it. Thanks. Will commit the changes as soon as possible

**Comment by neel1998** [comment]
@Avjeet  @cweitat  kindly review

**Comment by neel1998** [comment]
@Avjeet  @abhinavraj23  kindly review. This has been pending for a long time

**Comment by CloudyPadmal** [comment]
Conflicts to resolve

**Comment by neel1998** [comment]
@CloudyPadmal  i have resolved the conflicts. Kindly review

**Pull request merged**